### ISSUE TITLE
traefik v2 re-creates everything upon swarmRefresh

### ISSUE THREAD

**Issue by perosb** [containous/traefik#5419]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

upon swarmModeRefresh, traefik is recreating all routes, services, middlewares and resets health checks.

```time="2019-09-18T06:03:18Z" level=debug msg="Creating middleware" entryPointName=uu middlewareName=metrics-entrypoint middlewareType=Metrics
time="2019-09-18T06:03:18Z" level=debug msg="Creating middleware" entryPointName=traefik middlewareName=metrics-entrypoint middlewareType=Metrics
time="2019-09-18T06:03:18Z" level=debug msg="Creating middleware" middlewareName=metrics-entrypoint middlewareType=Metrics entryPointName=http
time="2019-09-18T06:03:18Z" level=debug msg="Initial health check for backend: \"oo@docker\""
time="2019-09-18T06:03:18Z" level=debug msg="Initial health check for backend: \"pp@docker\""
time="2019-09-18T06:03:18Z" level=debug msg="Creating TCP server 0 at 10.0.2.228:873" serverName=0 entryPointName=uu routerName=rsyncd@docker serviceName=uu
time="2019-09-18T06:03:18Z" level=debug msg="Creating TCP server 1 at 10.0.2.223:873" routerName=uu@docker serviceName=uu serverName=1 entryPointName=uu
```

### What did you expect to see?
I would have expected to see similar log as in v1.7
https://github.com/containous/traefik/blob/56e0580aa5ef5fc40a1969ec78014fef693c8a09/pkg/server/server_configuration.go#L220
```
evel=info msg="Skipping same configuration for provider docker"
```

### Output of `traefik version`: (_What version of Traefik are you using?_)
v2.0.0

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
running docker 19.x in swarm mode.

community post here: https://community.containo.us/t/docker-swarmmode-refresh/1659

**Comment by mpl** [comment]
@perosb We have reproduced the behaviour and identified 2 potential causes. one of them is a configuration issue (on which I'll elaborate later today), and the other one is a bug, for which we'll have a fix soon. Stay tuned.

**Comment by mpl** [comment]
@perosb See #5485 for the bugfix.
Now on to the configuration issue I was mentioning:
I don't know much about about docker swarm, but I noticed that even when nothing changes, some of the servers URLs in the dynamic config would change. As if maybe docker swarm would flip our containers onto different networks, or load-balance the app onto different containers, or something like that.
@ldez suggested to counter that by purposefully setting the network of the service, and that seemed to do the trick, i.e. the servers URLs would stop fluctuating. One way to achieve that is to set the traefik.docker.network label for your docker service.

**Comment by perosb** [comment]
Thanks @mpl.
I believe in my setup I already have all networks set.
All services have `traefik` as network as well as in the static config.
All stacks are running perfectly in 1.7 which is able to compare the refreshed config with the running.

So I'm "hoping" this caused by the issue you found.
I'm happy to beta test it if you like.

**Comment by mpl** [comment]
@perosb I'm hopeful the fix gets merged soon, but of course feel free to pull the change yourself and rebuild traefik with it (make build-image) if you like to live on the edge ;)

**Comment by traefiker** [comment]
Closed by #5485.
